python/ray/data/_internal/block_batching/block_batching.py

```diff
@@ -62,6 +64,8 @@ def batch_block_refs(
             ``pyarrow.Table``. Default is "default".
         drop_last: Whether to drop the last batch if it's incomplete.
         collate_fn: A function to apply to each data batch before returning it.
+        finalize_fn: A function to apply to each data batch after it has been collated.
+            This is executed on a GPU-based threadpool if available.
```

Comment:
```suggestion
            The prefetch depth for finalize_fn is always 1, so it can be used for heavyweight
            operations such as GPU preloading.
```